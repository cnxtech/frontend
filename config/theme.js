import theme from '@emcasa/ui'
import {fas} from '@fortawesome/pro-solid-svg-icons'
import {fal} from '@fortawesome/pro-light-svg-icons'
import icons from './icons'

export default {
  ...theme,
  icons: {emcasa: icons, default: fas, light: fal}
}
