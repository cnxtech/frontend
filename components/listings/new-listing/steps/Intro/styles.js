import styled from 'styled-components'
import theme from '@emcasa/ui'
import {breakpoint} from '@emcasa/ui/lib/styles'
import View from '@emcasa/ui-dom/components/View'
import Text from '@emcasa/ui-dom/components/Text'
import Row from '@emcasa/ui-dom/components/Row'

export const Space = styled(View)`
  display: block;
  height: 15vh;
  width: 1px;

  @media ${breakpoint.down('tablet')} {
    height: 5vh;
  }
`

export const Title = Text.withComponent('h1')

export const Background = styled(Row)`
  display: flex;
  width: 100vw;
  overflow: hidden;
  justify-content: center;
  background-image: url(https://res.cloudinary.com/emcasa/image/upload/fl_any_format/v1542831121/background/wall-small);
  background-size: cover;
  background-repeat: no-repeat;
  background-position: bottom;
`
