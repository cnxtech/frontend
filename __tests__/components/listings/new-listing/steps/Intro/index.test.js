import React from 'react';
import Intro from 'components/listings/new-listing/steps/Intro';
import renderer from 'react-test-renderer';
import 'jest-styled-components'
import theme from '@emcasa/ui'
import { ThemeProvider } from 'styled-components'

it('renders correctly', () => {
  const tree = renderer.create(
    <ThemeProvider theme={theme}>
      <Intro />
    </ThemeProvider>).toJSON()
  expect(tree).toMatchSnapshot()
})
