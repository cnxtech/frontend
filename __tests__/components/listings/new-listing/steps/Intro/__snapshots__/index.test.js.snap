// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 20px;
  padding-top: 0px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 20px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c8 {
  outline: none;
  display: block;
  cursor: pointer;
  padding: 0 20px 0 20px;
  border-color: #f50057;
  background-color: #f50057;
  border-radius: 4px;
  border: 1px solid;
  width: 100%;
  height: 60px;
  color: #ffffff;
  font-family: FaktSoftPro-Normal;
  font-size: 18px;
  -webkit-font-smoothing: antialiased;
  border-color: #e0e6ed;
}

.c8 :hover {
  background-color: #f50057;
}

.c1 {
  width: 100%;
}

.c4 {
  margin-bottom: 10px;
}

.c2 {
  -webkit-font-smoothing: antialiased;
  font-family: FaktSoftPro-Normal;
  color: #3c4858;
  font-size: 24px;
  font-weight: bold;
  text-align: center;
}

.c3 {
  -webkit-font-smoothing: antialiased;
  font-family: FaktSoftPro-Normal;
  color: #979797;
  font-size: 18px;
}

.c6 {
  -webkit-font-smoothing: antialiased;
  font-family: FaktSoftPro-Normal;
  color: #3c4858;
  font-size: 18px;
}

.c5 {
  margin-right: 10px;
}

@media screen and (min-width:780px) {
  .c1 {
    width: 50%;
  }
}

<div>
  <div
    className="c0"
  >
    <div
      className="c1"
      width={
        Array [
          1,
          0.5,
        ]
      }
    >
      <p
        className="c2"
        color="dark"
        fontSize="large"
        fontWeight="bold"
      >
        Como funciona vender seu imóvel aqui na EmCasa?
      </p>
      <p
        className="c3"
        color="grey"
        fontSize="default"
      >
        Venda seu imóvel de um jeito fácil e seguro.
      </p>
      <div
        className="c4"
      >
        <div
          className="c5"
          style={
            Object {
              "display": "inline",
              "verticalAlign": "sub",
            }
          }
        >
          <svg
            height={20}
            version="1.1"
            viewBox="0 0 512 512"
            x={0}
            y={0}
          >
            <path
              d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zm113.9 231L234.4 103.5c-9.4-9.4-24.6-9.4-33.9 0l-17 17c-9.4 9.4-9.4 24.6 0 33.9L285.1 256 183.5 357.6c-9.4 9.4-9.4 24.6 0 33.9l17 17c9.4 9.4 24.6 9.4 33.9 0L369.9 273c9.4-9.4 9.4-24.6 0-34z"
              fill="#1fb6ff"
              mr={2}
              name="chevron-circle-right"
              type="default"
            />
          </svg>
        </div>
        <span
          className="c6"
          color="dark"
          fontSize="default"
        >
          Cadastre seu imóvel no nosso site
        </span>
      </div>
      <div
        className="c4"
      >
        <div
          className="c5"
          style={
            Object {
              "display": "inline",
              "verticalAlign": "sub",
            }
          }
        >
          <svg
            height={20}
            version="1.1"
            viewBox="0 0 512 512"
            x={0}
            y={0}
          >
            <path
              d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zm113.9 231L234.4 103.5c-9.4-9.4-24.6-9.4-33.9 0l-17 17c-9.4 9.4-9.4 24.6 0 33.9L285.1 256 183.5 357.6c-9.4 9.4-9.4 24.6 0 33.9l17 17c9.4 9.4 24.6 9.4 33.9 0L369.9 273c9.4-9.4 9.4-24.6 0-34z"
              fill="#1fb6ff"
              mr={2}
              name="chevron-circle-right"
              type="default"
            />
          </svg>
        </div>
        <span
          className="c6"
          color="dark"
          fontSize="default"
        >
          Agende Tour Virtual 3D / Fotos grátis
        </span>
      </div>
      <div
        className="c4"
      >
        <div
          className="c5"
          style={
            Object {
              "display": "inline",
              "verticalAlign": "sub",
            }
          }
        >
          <svg
            height={20}
            version="1.1"
            viewBox="0 0 512 512"
            x={0}
            y={0}
          >
            <path
              d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zm113.9 231L234.4 103.5c-9.4-9.4-24.6-9.4-33.9 0l-17 17c-9.4 9.4-9.4 24.6 0 33.9L285.1 256 183.5 357.6c-9.4 9.4-9.4 24.6 0 33.9l17 17c9.4 9.4 24.6 9.4 33.9 0L369.9 273c9.4-9.4 9.4-24.6 0-34z"
              fill="#1fb6ff"
              mr={2}
              name="chevron-circle-right"
              type="default"
            />
          </svg>
        </div>
        <span
          className="c6"
          color="dark"
          fontSize="default"
        >
          Avaliação online do seu imóvel
        </span>
      </div>
      <div
        className="c4"
      >
        <div
          className="c5"
          style={
            Object {
              "display": "inline",
              "verticalAlign": "sub",
            }
          }
        >
          <svg
            height={20}
            version="1.1"
            viewBox="0 0 512 512"
            x={0}
            y={0}
          >
            <path
              d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zm113.9 231L234.4 103.5c-9.4-9.4-24.6-9.4-33.9 0l-17 17c-9.4 9.4-9.4 24.6 0 33.9L285.1 256 183.5 357.6c-9.4 9.4-9.4 24.6 0 33.9l17 17c9.4 9.4 24.6 9.4 33.9 0L369.9 273c9.4-9.4 9.4-24.6 0-34z"
              fill="#1fb6ff"
              mr={2}
              name="chevron-circle-right"
              type="default"
            />
          </svg>
        </div>
        <span
          className="c6"
          color="dark"
          fontSize="default"
        >
          Cuidamos de toda burocracia, contratos e documentação
        </span>
      </div>
      <div
        className="c4"
      >
        <div
          className="c5"
          style={
            Object {
              "display": "inline",
              "verticalAlign": "sub",
            }
          }
        >
          <svg
            height={20}
            version="1.1"
            viewBox="0 0 512 512"
            x={0}
            y={0}
          >
            <path
              d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zm113.9 231L234.4 103.5c-9.4-9.4-24.6-9.4-33.9 0l-17 17c-9.4 9.4-9.4 24.6 0 33.9L285.1 256 183.5 357.6c-9.4 9.4-9.4 24.6 0 33.9l17 17c9.4 9.4 24.6 9.4 33.9 0L369.9 273c9.4-9.4 9.4-24.6 0-34z"
              fill="#1fb6ff"
              mr={2}
              name="chevron-circle-right"
              type="default"
            />
          </svg>
        </div>
        <span
          className="c6"
          color="dark"
          fontSize="default"
        >
          Economize tempo e dinheiro anunciando seu imóvel com a gente
        </span>
      </div>
      <div
        className="c7"
      >
        <div
          className="c1"
          width={
            Array [
              1,
              0.5,
            ]
          }
        >
          <button
            className="c8"
            fontSize="default"
            height="tall"
            onClick={[Function]}
          >
            Quero Vender
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
